### ISSUE TITLE
pkg/unittest: test cases should be run in zones

### ISSUE THREAD

**Issue by DartBot** [dart-lang/test#228]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [kevmoo](https://github.com/kevmoo)**
_Originally opened as dart-lang/sdk#13277_

----

If each test case had its own Zone, then it would be straightforward to send async errors to the correct test case.

In theory, this could eliminate (or at least minimize) the use of guardAsync and handleExternalError

**Comment by DartBot** [comment]
<a href="https://github.com/floitschG"><img src="https://avatars.githubusercontent.com/u/8631949?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [floitschG](https://github.com/floitschG)**

----

Yes.
I think we should tests in their individual zones, once they are done.

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Marked this as being blocked by dart-lang/sdk#13334.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Marked this as being blocked by dart-lang/sdk#11279.*
*Unmarked this as being blocked by dart-lang/sdk#13334.*

**Comment by DartBot** [comment]
<a href="https://github.com/dgrove"><img src="https://avatars.githubusercontent.com/u/2108507?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [dgrove](https://github.com/dgrove)**

----

*Added [Triaged](../labels/Triaged) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/floitschG"><img src="https://avatars.githubusercontent.com/u/8631949?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [floitschG](https://github.com/floitschG)**

----

Issue dart-lang/test#210 has been merged into this issue.
______
 cc @gramster.
 cc @justinfagnani.

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

Committed https://code.google.com/p/dart/source/detail?r=32284
...and reverted: https://code.google.com/p/dart/source/detail?r=32288

Stay tuned
______
*Set owner to @kevmoo.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Marked this as being blocked by dart-lang/sdk#16541.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

https://codereview.chromium.org/133313004/

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

 dart-lang/sdk@b565b6296fcfae453a9fcdcc857c03e19faf65a6
______
*Added [Fixed](../labels/Fixed) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Added [Pkg-Unittest](../labels/Pkg-Unittest) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Removed [Area-UnitTest](../labels/Area-UnitTest) label.*
*Added [Area-Pkg](../labels/Area-Pkg) label.*
